### PR TITLE
feat(threaded-replies): Create Reply button component

### PR DESCRIPTION
**Pull request by JChan106**
This PR includes the CreateReply component which will be used in the Replies component in another PR. This component is a toggle which switches from a button to a Comment form based on the props.

- [x]  Unit Tests

### CODE REVIEW

File: src/elements/content-sidebar/activity-feed/comment-form/CommentForm.js [hunk 1]
```diff
@@ -129,7 +129,7 @@ class CommentForm extends React.Component<Props, State> {
 
         return (
             <Media className={inputContainerClassNames}>
-                {!isEditing && (
+                {!isEditing && !!user && !!getAvatarUrl && (
```
Comment (karelee7) [inline src/elements/content-sidebar/activity-feed/comment-form/CommentForm.js:132]: Should we pull this logic out into a separate variable for readability?

File: src/elements/content-sidebar/activity-feed/comment-form/CommentForm.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ type Props = {
     contactsLoaded?: boolean,
     createComment?: Function,
     entityId?: string,
-    getAvatarUrl: GetAvatarUrlCallback,
+    getAvatarUrl?: GetAvatarUrlCallback,
```
Comment (karelee7) [inline src/elements/content-sidebar/activity-feed/comment-form/CommentForm.js:30]: Curious, what do we do when there is no avatar url? Do we show anything in place of the avatar?

File: src/elements/content-sidebar/activity-feed/comment/CreateReply.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
+// @flow
+
+import * as React from 'react';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import type { IntlShape } from 'react-intl';
+import noop from 'lodash/noop';
+
+import PlainButton from '../../../../components/plain-button';
+import ArrowArcRight from '../../../../icon/fill/ArrowArcRight';
```
Comment (karelee7) [inline src/elements/content-sidebar/activity-feed/comment/CreateReply.js:9]: minor: alphabetical order

File: src/elements/content-sidebar/activity-feed/comment/CreateReply.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import type { SelectorItems } from '../../../../common/types/core';
+
+import CommentForm from '../comment-form';
+
+import messages from './messages';
+import './CreateReply.scss';
+
+type Props = {
+    getMentionWithQuery?: (searchStr: string) => void,
+    intl: IntlShape,
```
Comment (karelee7) [inline src/elements/content-sidebar/activity-feed/comment/CreateReply.js:20]: Do we have to include this in the props? Could we just import `intl` to use below?

File: src/elements/content-sidebar/activity-feed/comment/__tests__/CreateReply.test.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    });
+
+    test('should correctly render CreateReply form', () => {
+        getWrapper({ showReplyForm: true });
+
+        expect(screen.getByText(replyInThreadMessage)).toBeVisible();
+        expect(screen.queryByText(replyMessage)).not.toBeInTheDocument();
+    });
+
+    test('should disable Reply button if isDisabled property is true', () => {
```
Comment (karelee7) [inline src/elements/content-sidebar/activity-feed/comment/__tests__/CreateReply.test.js:58]: Should we add a test for when `isDisabled` is false?

File: src/elements/content-sidebar/activity-feed/comment-form/CommentForm.js (hunk 1 above)
Comment (JChan106) [reply src/elements/content-sidebar/activity-feed/comment-form/CommentForm.js:132]: Tried doing that but it gives some linting errors

File: src/elements/content-sidebar/activity-feed/comment-form/CommentForm.js (hunk 2 above)
Comment (JChan106) [reply src/elements/content-sidebar/activity-feed/comment-form/CommentForm.js:30]: In this case we wouldn't show anything. Previously we would require the cb, but the reply form doesn't need an avatar so I made it optional, and to just not show the avatar if there's no cb passed in.

File: src/elements/content-sidebar/activity-feed/comment/__tests__/CreateReply.test.js (hunk 5 above)
Comment (JChan106) [reply src/elements/content-sidebar/activity-feed/comment/__tests__/CreateReply.test.js:58]: isDisabled is false by default.

File: src/elements/content-sidebar/activity-feed/comment-form/CommentForm.js (hunk 2 above)
Comment (JChan106) [reply src/elements/content-sidebar/activity-feed/comment-form/CommentForm.js:30]: Actually looks like `Avatar.js` doesn't require a `getAvatarUrl` in the component. So I will remove that boolean from the conditional.

File: src/elements/content-sidebar/activity-feed/comment/__tests__/CreateReply.test.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+        getWrapper();
+
+        fireEvent.click(screen.getByText(replyMessage));
+        expect(onClick).toBeCalledTimes(1);
+        expect(onSubmit).not.toBeCalled();
+        expect(onFocus).not.toBeCalled();
+        expect(onCancel).not.toBeCalled();
+    });
+
+    test('should not show form when isDisabled', () => {
```
Comment (karelee7) [inline src/elements/content-sidebar/activity-feed/comment/__tests__/CreateReply.test.js:78]: minor: `should not show form when isDisabled` -> `should not show form when isDisabled is true`

Comment (karelee7) [inline src/elements/content-sidebar/activity-feed/comment/__tests__/CreateReply.test.js:78]: When `isDisabled` is false, do we show this form? If so, do we need a test for that??

Comment (JChan106) [reply src/elements/content-sidebar/activity-feed/comment/__tests__/CreateReply.test.js:78]: there's a test for that already, when you don't have the `isDisabled` as a prop it defaults to false.

**Pull request merged**